models/classes/class.QtiTestService.php

```diff
@@ -866,24 +869,62 @@ protected function importTest(
         return $report;
     }
 
+    /**
+     * @throws common_Exception
+     */
     private function deleteTestsFromClassByLabel(
         string $testLabel,
         string $itemsClassLabel,
         core_kernel_classes_Resource $testClass,
         core_kernel_classes_Class $itemClass
     ): void {
-        $testService = $this->getTestService();
-        $itemTreeService = $this->getItemTreeService();
+        $itemClassesToDelete = $this->getSubclassesByLabel($itemClass, $itemsClassLabel);
+
+        /** @var string[] $refs */
+        $refs = $this->collectResourceReferences($itemClassesToDelete);
 
+        /** @var string[] $deletedTestsUris */
+        $deletedTestsUris = [];
         foreach ($testClass->getInstances() as $testInstance) {
+            /** @var core_kernel_classes_Resource $testInstance */
             if ($testInstance->getLabel() === $testLabel) {
-                $testService->deleteTest($testInstance);
+                $deletedTestsUris[] = $testInstance->getUri();
             }
         }
 
-        foreach ($itemClass->getSubClasses() as $subClass) {
-            if ($subClass->getLabel() === $itemsClassLabel) {
-                $itemTreeService->deleteClass($subClass);
+        /** @var string[] $deletedItemClassesUris */
+        $deletedItemClassesUris = [];
```

Comment:
Why type hint local variables? Is it sustainable to do it?

```suggestion
        $deletedItemClassesUris = [];
```

Reply:
Thanks!